File: src/array/1630.cpp
class Solution
{
public:
  vector<bool> checkArithmeticSubarrays(vector<int> &nums, vector<int> &l, vector<int> &r)
  {
    vector<bool> result;

    for (int i = 0; i < l.size(); i++)
    {
      vector<int> arr;
      for (int j = l[i]; j <= r[i]; j++)
      {
        arr.push_back(nums[j]);
      }
      result.push_back(check(arr));
    }

    return result;
  }
  bool check(vector<int> arr)
  {
    if (arr.size() <= 2)
    {
      return true;
    }
    sort(arr.begin(), arr.end());
    int diff = arr[1] - arr[0];
    for (int i = 2; i < arr.size(); i++)
    {
      if (arr[i] - arr[i - 1] != diff)
      {
        return false;
      }
    }
    return true;
  }
};

File: src/array/1630.js
/**
 * 检测数组是否是等差数列
 * @param {number[]} arr
 * @returns
 */
const check = (arr) => {
  const { length } = arr;
  if (length <= 2) {
    return true;
  }
  arr.sort((a, b) => (a < b ? -1 : 1));
  const diff = arr[1] - arr[0];

  for (let i = 2; i < length; i++) {
    if (arr[i] - arr[i - 1] !== diff) {
      return false;
    }
  }
  return true;
};

/**
 * @param {number[]} nums
 * @param {number[]} l
 * @param {number[]} r
 * @return {boolean[]}
 */
var checkArithmeticSubarrays = function (nums, l, r) {
  return l.map((left, index) => {
    const right = r[index];
    const arr = [];

    for (let i = left; i <= right; i++) {
      arr.push(nums[i]);
    }
    return check(arr);
  });
};
